src/lib.rs

```diff
@@ -203,13 +213,22 @@ fn update_state<T: AtLeastF32>(
     frac_orthogonal: &mut T,
     time_parallel: &T,
 ) {
-    if *velocity_parallel >= 0.0.into() {
-        *coord_parallel += 1;
-        *frac_parallel = 0.0.into();
-    } else {
-        *coord_parallel -= 1;
-        *frac_parallel = 1.0.into();
-    }
+    // branchless version of
+    // if *velocity_parallel >= 0.0.into() {
+    //     *coord_parallel += 1;
+    // } else {
+    //     *coord_parallel -= 1;
+    // }
+    *coord_parallel += signum(*velocity_parallel).round_to_i64();
```

Comment:
actually this may be considered *more* correct than the code it replaces in the case where `velocity_parallel == 0.0`, so it could be argued that this works as a bug fix, in which case, it doesn't need to improve perfs (but it would still be desirable that it doesn't hurt either)